code/API_definitions/device-status.yaml

```diff
@@ -79,8 +110,167 @@ paths:
           $ref: "#/components/responses/Generic500"
         '503':
           $ref: "#/components/responses/Generic503"
-        '504':
-          $ref: "#/components/responses/Generic504"
+  /roaming/event-subscription:
+    post:
+      tags:
+        - Device roaming status subscription
+      summary: 'Create a device roaming status subscription for a device'
+      operationId: createRoamingSubscription
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateRoamingSubscription'
+        required: true
+      responses:
+        '201':
```

Comment:
I think that event subscription should be performed synchronously, and if the server times out while waiting for the request to finish, we can respond with a 408 error code.

Reply:
I'm on same page than @sachinvodafone that the subscription must be sync. For the error I propose to stick to the guidelines (I will make sure to have all) 
for POST: 400, 401, 403, 409, 500, 503
for GET: 400, 401, 403, 500, 503
for GET/{subscriptionId}: 400, 401, 403, 404, 500, 503
for DELETE: 400, 401, 403, 404, 500, 503

Reply:
But we could allow both, couldn't we? Why do you think we should force them to be synchronous and not allow to choose?

Reply:
@monamok  But why you want this for this particular operation? 
The event subscription is a standard request not related by asynchronous mechanism itself. If we add this, we should also add to POST / roaming  and all other POST operation.
The only exception for me is the one for the notification itself.

Reply:
While it is possible to implement asynchronous operations for both GET and POST operations across all endpoints in the Camara API, it's important to note that certain operations may inherently require immediate responses. These operations are better suited for synchronous handling. It is necessary for us to align and agree on this whether it is best suited for Syn or Asy , as this consideration will apply to every use case.

Reply:
I understand the point of view of having synchronous response in create/deleteSubscriptions. 
Having said that, during notification guidelines track, as per PR https://github.com/camaraproject/WorkingGroups/pull/173, the response codes for successful operations were not discussed. Thus no decision was made about including only synchronous operations or we could include both mechanisms for “write” subscription operations (in case of this API. POST -createSubscription-, DELETE -deleteSubscription-). In subscription section of the guideline specifies the error codes for eachpoint, but not the success codes. There are some examples given for success cases, but other than that, no clear instruction which says we can't consider asynchronous responses.

On the other hand, in https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md#32-http-response-codes, it is reflected the use of 202 Accepted for asynchronous process.

In Telefonica, an asynchronous-based subscription model is designed, because this “write” operations manage actions in several underlying systems and that could take some time. For that reason, we would like to support both:

- 201 and 202 for create (we are returning required parameter eventSubscriptionId anyway)
- 202 and 204 for delete 

In case we allow 202 responses, we should mention in the guideline that GET /subscriptions/{subscriptionId} must be used to ensure existence of the subscription.